tests/subsys/power/power_mgmt/subsys.power.device_pm  fails to build on nrf5340dk_nrf5340_cpunet &  nrf5340pdk_nrf5340_cpunet

```
scripts/twister -p nrf5340dk_nrf5340_cpunet -p nrf5340pdk_nrf5340_cpunet -s tests/subsys/power/power_mgmt/subsys.power.device_pm
```

Gets:

```
FAILED: CMakeFiles/app.dir/src/main.c.obj 
ccache /opt/sdk/zephyr-sdk-0.12.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=v2.5.0-rc1-16-g085d2eda8d2b -DKERNEL -DNRF5340_XXAA_NETWORK -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR__=1 -I/workdir/zephyr/kernel/include -I/workdir/zephyr/arch/arm/include -I/workdir/zephyr/include -Izephyr/include/generated -I/workdir/zephyr/soc/arm/nordic_nrf/nrf53 -I/workdir/zephyr/subsys/testsuite/include -I/workdir/zephyr/subsys/testsuite/ztest/include -I/var/lib/buildkite-agent/zephyr-module-cache/modules/hal/cmsis/CMSIS/Core/Include -I/workdir/modules/hal/nordic/nrfx -I/workdir/modules/hal/nordic/nrfx/drivers/include -I/workdir/modules/hal/nordic/nrfx/mdk -I/workdir/zephyr/modules/hal_nordic/nrfx/. -isystem /workdir/zephyr/lib/libc/minimal/include -isystem /opt/sdk/zephyr-sdk-0.12.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.2.0/include -isystem /opt/sdk/zephyr-sdk-0.12.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.2.0/include-fixed -Os -imacros /workdir/zephyr/twister-out/nrf5340pdk_nrf5340_cpunet/tests/subsys/power/power_mgmt/subsys.power.device_pm/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m33+nodsp -mthumb -mabi=aapcs -imacros /workdir/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wno-address-of-packed-member -Wno-unused-but-set-variable -Werror=implicit-int -Werror -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/workdir/zephyr/tests/subsys/power/power_mgmt=CMAKE_SOURCE_DIR -fmacro-prefix-map=/workdir/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/workdir=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT CMakeFiles/app.dir/src/main.c.obj -MF CMakeFiles/app.dir/src/main.c.obj.d -o CMakeFiles/app.dir/src/main.c.obj   -c /workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c: In function 'pm_policy_low_power_devices':
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:64:9: error: implicit declaration of function 'pm_is_sleep_state' [-Werror=implicit-function-declaration]
   64 |  return pm_is_sleep_state(state);
      |         ^~~~~~~~~~~~~~~~~
In file included from /workdir/zephyr/subsys/testsuite/ztest/include/ztest.h:60,
                 from /workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:10:
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c: In function 'pm_policy_next_state':
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:74:31: error: 'struct z_kernel' has no member named 'idle'
   74 |  zassert_true(ticks == _kernel.idle, NULL);
      |                               ^
/workdir/zephyr/subsys/testsuite/ztest/include/ztest_assert.h:93:12: note: in definition of macro 'zassert'
   93 |  z_zassert(cond, msg ? ("(" default_msg ")") : (default_msg), \
      |            ^~~~
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:74:2: note: in expansion of macro 'zassert_true'
   74 |  zassert_true(ticks == _kernel.idle, NULL);
      |  ^~~~~~~~~~~~
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c: In function 'test_main':
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:214:9: error: variable 'notifier' has initializer but incomplete type
  214 |  struct pm_notifier notifier = {
      |         ^~~~~~~~~~~
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:215:4: error: 'struct pm_notifier' has no member named 'state_entry'
  215 |   .state_entry = notify_pm_state_entry,
      |    ^~~~~~~~~~~
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:215:18: error: excess elements in struct initializer [-Werror]
  215 |   .state_entry = notify_pm_state_entry,
      |                  ^~~~~~~~~~~~~~~~~~~~~
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:215:18: note: (near initialization for 'notifier')
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:216:4: error: 'struct pm_notifier' has no member named 'state_exit'
  216 |   .state_exit = notify_pm_state_exit,
      |    ^~~~~~~~~~
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:216:17: error: excess elements in struct initializer [-Werror]
  216 |   .state_exit = notify_pm_state_exit,
      |                 ^~~~~~~~~~~~~~~~~~~~
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:216:17: note: (near initialization for 'notifier')
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:214:21: error: storage size of 'notifier' isn't known
  214 |  struct pm_notifier notifier = {
      |                     ^~~~~~~~
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:219:2: error: implicit declaration of function 'pm_notifier_register' [-Werror=implicit-function-declaration]
  219 |  pm_notifier_register(&notifier);
      |  ^~~~~~~~~~~~~~~~~~~~
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:231:2: error: implicit declaration of function 'pm_notifier_unregister' [-Werror=implicit-function-declaration]
  231 |  pm_notifier_unregister(&notifier);
      |  ^~~~~~~~~~~~~~~~~~~~~~
/workdir/zephyr/tests/subsys/power/power_mgmt/src/main.c:214:21: error: unused variable 'notifier' [-Werror=unused-variable]
  214 |  struct pm_notifier notifier = {
      |                     ^~~~~~~~
cc1: all warnings being treated as errors
[16/142] Building C object zephyr/CMakeFiles/zephyr.dir/li
```

this boils down to this board failing when CONFIG_PM is enabled (which enables build of `soc/arm/nordic_nrf/nrf53/power.c` with the following failure:

```
/home/nashif/Work/zephyrproject/modules/hal/nordic/nrfx/hal/nrf_regulators.h:138:50: error: unknown type name 'NRF_REGULATORS_Type'; did you mean 'NRF_REGULATORS_H__'?
  138 | NRF_STATIC_INLINE void nrf_regulators_dcdcen_set(NRF_REGULATORS_Type * p_reg, bool enable);
      |                                                  ^~~~~~~~~~~~~~~~~~~
      |                                                  NRF_REGULATORS_H__
/home/nashif/Work/zephyrproject/modules/hal/nordic/nrfx/hal/nrf_regulators.h:150:50: error: unknown type name 'NRF_REGULATORS_Type'; did you mean 'NRF_REGULATORS_H__'?
  150 | NRF_STATIC_INLINE void nrf_regulators_system_off(NRF_REGULATORS_Type * p_reg);
      |                                                  ^~~~~~~~~~~~~~~~~~~
      |                                                  NRF_REGULATORS_H__
/home/nashif/Work/zephyrproject/modules/hal/nordic/nrfx/hal/nrf_regulators.h:234:50: error: unknown type name 'NRF_REGULATORS_Type'; did you mean 'NRF_REGULATORS_H__'?
  234 | NRF_STATIC_INLINE void nrf_regulators_dcdcen_set(NRF_REGULATORS_Type * p_reg, bool enable)
      |                                                  ^~~~~~~~~~~~~~~~~~~
      |                                                  NRF_REGULATORS_H__
/home/nashif/Work/zephyrproject/modules/hal/nordic/nrfx/hal/nrf_regulators.h:243:50: error: unknown type name 'NRF_REGULATORS_Type'; did you mean 'NRF_REGULATORS_H__'?
  243 | NRF_STATIC_INLINE void nrf_regulators_system_off(NRF_REGULATORS_Type * p_reg)
      |                                                  ^~~~~~~~~~~~~~~~~~~
      |                                                  NRF_REGULATORS_H__
/home/nashif/Work/zephyrproject/zephyr/soc/arm/nordic_nrf/nrf53/power.c: In function 'pm_power_state_set':
/home/nashif/Work/zephyrproject/zephyr/soc/arm/nordic_nrf/nrf53/power.c:20:3: warning: implicit declaration of function 'nrf_regulators_system_off' [-Wimplicit-function-declaration]
   20 |   nrf_regulators_system_off(NRF_REGULATORS);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/nashif/Work/zephyrproject/zephyr/soc/arm/nordic_nrf/nrf53/power.c:20:29: error: 'NRF_REGULATORS' undeclared (first use in this function); did you mean 'NRF_REGULATORS_H__'?
   20 |   nrf_regulators_system_off(NRF_REGULATORS);
      |                             ^~~~~~~~~~~~~~
      |                             NRF_REGULATORS_H__
/home/nashif/Work/zephyrproject/zephyr/soc/arm/nordic_nrf/nrf53/power.c:20:29: note: each undeclared identifier is reported only once for each function it appears in
```

Should building of power.c in the SoC depend on something else to prevent that or to correctly pull in regulator support? @pabigot  ?

--- a/soc/arm/nordic_nrf/Kconfig.defconfig
+++ b/soc/arm/nordic_nrf/Kconfig.defconfig
@@ -28,7 +28,7 @@ config ARCH_HAS_CUSTOM_BUSY_WAIT
 	default y
 
 config PM
-	default y if SYS_CLOCK_EXISTS
+	default y if SYS_CLOCK_EXISTS && !HAS_NO_SYS_PM
 
 config BUILD_OUTPUT_HEX
 	default y

--- a/soc/arm/nordic_nrf/Kconfig.defconfig
+++ b/soc/arm/nordic_nrf/Kconfig.defconfig
@@ -27,6 +27,9 @@ config SYS_CLOCK_TICKS_PER_SEC
 config ARCH_HAS_CUSTOM_BUSY_WAIT
 	default y
 
+config PM
+	default y if SYS_CLOCK_EXISTS
+
 config BUILD_OUTPUT_HEX
 	default y
 

--- a/soc/arm/nordic_nrf/nrf53/Kconfig.soc
+++ b/soc/arm/nordic_nrf/nrf53/Kconfig.soc
@@ -96,6 +96,7 @@ config SOC_NRF5340_CPUNET
 	select HAS_HW_NRF_TWIS0
 	select HAS_HW_NRF_UARTE0
 	select HAS_HW_NRF_WDT
+	select HAS_NO_SYS_PM
 
 choice
 	prompt "nRF53x MCU Selection"

--- a/subsys/power/Kconfig
+++ b/subsys/power/Kconfig
@@ -13,7 +13,7 @@ config SYS_POWER_MANAGEMENT
 menuconfig PM
 	bool "System Power management"
 	select TICKLESS_IDLE
-	depends on SYS_CLOCK_EXISTS && !SOC_NRF5340_CPUNET_QKAA
+	depends on SYS_CLOCK_EXISTS
 	help
 	  This option enables the board to implement extra power management
 	  policies whenever the kernel becomes idle. The kernel informs the

--- a/subsys/power/Kconfig
+++ b/subsys/power/Kconfig
@@ -13,7 +13,7 @@ config SYS_POWER_MANAGEMENT
 menuconfig PM
 	bool "System Power management"
 	select TICKLESS_IDLE
-	depends on SYS_CLOCK_EXISTS
+	depends on SYS_CLOCK_EXISTS && !HAS_NO_SYS_PM
 	help
 	  This option enables the board to implement extra power management
 	  policies whenever the kernel becomes idle. The kernel informs the
@@ -51,6 +51,14 @@ source "subsys/logging/Kconfig.template.log_config"
 
 
 endif # PM
+
+config HAS_NO_SYS_PM
+	bool
+	help
+	  This option blocks selection of PM.  It can be selected in SOC
+	  targets where system power management is not supported, for example
+	  on support core of a multi-core device where SoC power management is
+	  the responsibility of a different core.
 
 config PM_DEVICE
 	bool "Device power management"

--- a/tests/kernel/profiling/profiling_api/testcase.yaml
+++ b/tests/kernel/profiling/profiling_api/testcase.yaml
@@ -1,5 +1,7 @@
 tests:
   kernel.common.profiling:
     arch_exclude: nios2
-    platform_exclude: em_starterkit litex_vexriscv rv32m1_vega_zero_riscy rv32m1_vega_ri5cy
+    platform_exclude: em_starterkit
+      litex_vexriscv rv32m1_vega_zero_riscy rv32m1_vega_ri5cy
+      nrf5340dk_nrf5340_cpunet nrf5340pdk_nrf5340_cpunet
     tags: kernel

--- a/tests/kernel/tickless/tickless_concept/testcase.yaml
+++ b/tests/kernel/tickless/tickless_concept/testcase.yaml
@@ -3,5 +3,7 @@ tests:
     arch_exclude: nios2
     # FIXME: This test fails sporadically on all QEMU platforms, but fails
     # consistently when coverage is enabled. Disable until 14173 is fixed.
-    platform_exclude: qemu_x86_coverage litex_vexriscv rv32m1_vega_zero_riscy rv32m1_vega_ri5cy
+    platform_exclude: qemu_x86_coverage
+      litex_vexriscv rv32m1_vega_zero_riscy rv32m1_vega_ri5cy
+      nrf5340dk_nrf5340_cpunet nrf5340pdk_nrf5340_cpunet
     tags: kernel

--- a/tests/kernel/timer/timer_api/testcase.yaml
+++ b/tests/kernel/timer/timer_api/testcase.yaml
@@ -5,5 +5,7 @@ tests:
   kernel.timer.tickless:
     extra_args: CONF_FILE="prj_tickless.conf"
     arch_exclude: nios2 posix
-    platform_exclude: qemu_x86_coverage litex_vexriscv rv32m1_vega_zero_riscy rv32m1_vega_ri5cy
+    platform_exclude: qemu_x86_coverage
+      litex_vexriscv rv32m1_vega_zero_riscy rv32m1_vega_ri5cy
+      nrf5340dk_nrf5340_cpunet nrf5340pdk_nrf5340_cpunet
     tags: kernel timer userspace

--- a/tests/subsys/power/power_mgmt/testcase.yaml
+++ b/tests/subsys/power/power_mgmt/testcase.yaml
@@ -2,8 +2,11 @@ tests:
   subsys.power.device_pm:
     # arch_irq_unlock(0) can't work correctly on these arch
     arch_exclude: arc xtensa
-    # When CONFIG_TICKLESS_IDLE enable, these platforms don't provide timer driver
+    # When CONFIG_TICKLESS_IDLE enable, some of these platforms don't
+    # provide a timer driver.  Other platforms are excluded because
+    # they lack some other required feature.
     platform_exclude: rv32m1_vega_ri5cy rv32m1_vega_zero_riscy litex_vexriscv
+       nrf5340dk_nrf5340_cpunet nrf5340pdk_nrf5340_cpunet
     integration_platforms:
       - qemu_x86
       - mps2_an385

